sql/core/src/test/scala/org/apache/spark/sql/CollationStringExpressionsSuite.scala

```diff
@@ -163,6 +163,169 @@ class CollationStringExpressionsSuite
     })
   }
 
+  test("substring check output type on explicitly collated string") {
```

Comment:
Let's try to unify test naming

e.g. "Support Substring string expression with collation"